docs/DeveloperGuide.md

```diff
@@ -1,38 +1,160 @@
 
-## Design & implementation
 
+###### [Back to table of contents](#table-of-contents)
+
+## Design & Implementation
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Architecture
+
+###### [Back to table of contents](#table-of-contents)
+
+We aim to help fellow gym enthusiasts and new gym members to help keep
+track of their future and present training.
+=======
+### UI component
+
+###### [Back to table of contents](#table-of-contents)
+
+### Start component
+
+###### [Back to table of contents](#table-of-contents)
+
+### Add component
+
+The add mechanism is facilitated by `AddCommand`. 
+It extends `Command` and modifies the execute function to add an exercise.
+
+<img src="images/AddExerciseDiagram.png" width="450" />
+
+Given below is an example usage scenario and how the add mechanism behaves at each step.
+
+Step 1. The user enters the add command with the necessary arguments.
+
+Step 2. The input is processed by the `Parser` to separate out the arguments and creates the exercise to be added, 
+`toAdd`.
+
+Step 3. The `AddCommand` calls `execute()` which calls `WorkList#getCurrentWorkout()` to return `currentWorkout`.
+
+Step 4. Finally, `addExercise()` is called and `toAdd` is added to `currentWorkout`.
+
+###### [Back to table of contents](#table-of-contents)
 
-## Product scope
-### Target user profile
+### List component
 
-{Describe the target user profile}
+###### [Back to table of contents](#table-of-contents)
 
-### Value proposition
+### View component
+The View component is facilitated by `Parser`,`Ui`,`WorkoutList`,`Command` and `ViewCommand`, where the user will 
+enter a specific workout date and the number of exercises on that date will be displayed
+
+Below are the specific steps on how to use the view function and how the mechanism will flow:
+
+* Step 1: We will assume that the user has started a workout on two specific dates, `11/02/22` which was added with the following command `/start 11/02/22`
+  and `12/02/22`, which was added with the following command `/start 12/02/22`.
+* Step 2: The user will then use the following command `/view 11/02/22` will be taken into the parser
+  and will return a list of exercises done on that specified date.
+
+###### [Back to table of contents](#table-of-contents)
+
+### Delete component
+The deletion mechanism is facilitated by 'Parser', 'DeleteCommand', 'Workout', 'WorkoutList' and 'UI', where a Workout object will be deleted according to the command inputted by the user and removed from the workout list.
+
+Below is an example usage scenario and how the deletion mechanism behaves at each step:
+
+Step 1: Assume that the user has already added a workout on 21/03/23 into the WorkoutList using the following command, /start 21/03/23
+
+Step 2: The user input of /delete 21/03/23 will be taken in for the parser and an object of class DeleteCommand will be returned.
```

Comment:
Perhaps you can use the Markdown format for code snippets, such as for `/delete 21/03/23` so that the explanation can be differentiated from the code syntax